supervision/draw/utils.py

```diff
@@ -171,3 +172,58 @@ def draw_text(
         lineType=cv2.LINE_AA,
     )
     return scene
+
+
+def copy_paste(
+    source_image: np.ndarray,
+    source_polygon: np.ndarray,
+    target_image: np.ndarray,
+    scale: float = 1.0,
+    x: int = 0,
+    y: int = 0,
+):
+    """
+    Copy and paste a region from a source image into a target image.
+
+    Attributes:
+        source_image (np.ndarray): The image to be copied.
+        source_polygon (np.ndarray): The polygon area to be copied.
+        target_image (np.ndarray): The image to be pasted into.
+        scale: scale factor to apply to the source image
+        x: x offset to paste into the target image at
+        y: y offset to paste into the target image at
+
+
+    Returns:
+        np.ndarray: The target image with the source image pasted into it.
+    """
+    # cv2.imshow('source', source_image)
+    # cv2.imshow('target', target_image)
+    # # print(source_image.shape)
+
+    output = target_image.copy()
+
+    # generate mask image based on polygon
+    source_width = source_image.shape[0]
+    source_height = source_image.shape[1]
+    mask = generate_2d_mask(source_polygon, (source_width, source_height))
+
+    # cutout = cv2.bitwise_and(self.crop, self.crop, mask=self.mask)
+
+    # scale the source and mask
```

Comment:
The order of operations makes it inefficient.

1. Scale source <- expensive
2. Scale mask <- expensive
3. Extract patch from source <- cheap

Let's refactor and make it:

1. Extract patch from source <- cheap
2. Scale patch <- expensive

Reply:
mhh, I see what you're saying but I don't think that works with the current approach.

The `patch` is an image with both source image pixels and the background pixels from the target area.  So if we want to scale the patch, we would have to upscale the background first to generate the patch.  Maybe there is a better way to do the pasting / merging of pixels instead of this?:
```
    # generate a patch from the source image  / with background from target image where we are pasting
    patch = np.where(
        np.expand_dims(scaled_mask, axis=2),
        scaled_source,
        target_image[y : y + scaled_height, x : x + scaled_width, :],
    )
```

Reply:
I think I'm skipping the step of extracting just a section of the source image (for the magic scissors use case its already cropped to bounding box of polygon)....so actually I think maybe want to add that part first, and then scale the section extracted/cropped from source.

Reply:
Oh, so the `patch` has the same shape as the target image? So if the image is 4K and the object we are copying is `400x400` we will still create a 4K patch?

Don't get me wrong, I can merge it as it is, but it sounds like something we could optimize quite fast. Because that line:

```
patch = np.where(
    np.expand_dims(scaled_mask, axis=2),
    scaled_source,
    target_image[y : y + scaled_height, x : x + scaled_width, :],
)
```

Is essentially a hidden image copy.